lib/shoryuken/extensions/active_job_adapter.rb

```diff
@@ -75,6 +75,16 @@ class JobWrapper #:nodoc:
         shoryuken_options body_parser: :json, auto_delete: true
 
         def perform(sqs_msg, hash)
+          approximate_receive_count = sqs_msg.attributes['ApproximateReceiveCount'].to_i
+          if approximate_receive_count > 1
+            message_queue_name = JSON.parse(sqs_msg.body)['queue_name']
+            redrive_policy_json = JSON.parse(Shoryuken::Client.queues(message_queue_name).redrive_policy)
```

Comment:
At startup you don't know all your queues. It's only at runtime you can start to infer this kind of information. Adding a msg to the queue and fetching a msg from the key are the only 2 moments when you have the queue name.

A queue has a redrive policy. Once set, it never changes until the queue is deleted and created again. I tried to update the redrive policy but that is not allowed.
In the updated code the policy is stored in a hash with as key the queue name. 
This is the best approach I could come up with at this point.

I'm really eager to get this issue sorted, because consuming a msg from the DLQ can be very useful. So, if you can give me a few hints on how you would solve it, that would be great.